x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -556,6 +558,14 @@ public Map<String, Processor.Factory> getProcessors(Processor.Parameters paramet
         return Collections.singletonMap(InferenceProcessor.TYPE, inferenceFactory);
     }
 
+    @Override
+    public void loadExtensions(ExtensionLoader loader) {
+        loader.loadExtensions(MachineLearningExtension.class).forEach(machineLearningExtension::set);
```

Comment:
All the ML `internalClusterTests` tests failed in CI. The reason is that those tests don't use the `MachineLearning` plugin, but a `LocalStateMachineLearning` plugin that wraps it: https://github.com/elastic/elasticsearch/blob/264f09f3d571d1311eaa1aee2af22723a61736a1/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/support/BaseMlIntegTestCase.java#L146

So, something needs to be done about that.

One way to fix it is probably to make this change:

```suggestion
        if (loader != null) {
            loader.loadExtensions(MachineLearningExtension.class).forEach(machineLearningExtension::set);
        }
```

and then also add a call to `mlPlugin.loadExtensions(null)` adjacent to https://github.com/elastic/elasticsearch/blob/fc793e1b7d13d889a763e22cec5dd5818463927b/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/LocalStateMachineLearning.java#L53